script_test.go

```diff
@@ -1286,6 +1286,39 @@ func TestFindFiles_InNonexistentPathReturnsError(t *testing.T) {
 	}
 }
 
+func TestFindFiles_DoesNotErrorWhenSubDirectoryIsNotReadable(t *testing.T) {
+	t.Parallel()
+	tmpDir := t.TempDir()
+	fileAPath := filepath.Join(tmpDir, "file_a.txt")
+	if err := os.WriteFile(fileAPath, []byte("hello world!"), os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	restrictedDirPath := filepath.Join(tmpDir, "restricted_dir")
+	if err := os.Mkdir(restrictedDirPath, os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	fileBPath := filepath.Join(restrictedDirPath, "file_b.txt")
+	if err := os.WriteFile(fileBPath, []byte("hello again!"), os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(restrictedDirPath, 0o000); err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() { os.Chmod(restrictedDirPath, os.ModePerm) })
```

Comment:
This tempdir will be deleted anyway.

Reply:
@bitfield I need to do this so that this dir can get cleaned up. Otherwise, the cleanup function throws a permissions error. Let me know if you'd like to do this a different way. 
```
TempDir RemoveAll cleanup: openfdat /var/folders/mn/31m4g_6j7qz26m7_00mgsbrc0000gn/T/TestFindFiles_DoesNotErrorWhenSubDirectoryIsNotReadable1455531535/001/restricted_dir: permission denied
```

Reply:
That's because you're also creating a file _inside_ the restricted directory. So `RemoveAll` isn't able to delete the file, because that requires modifying the directory, which you don't have permission to do.

You don't need this file for the test (and `FindFIles` won't be able to even know it exists, because of the directory permissions). Without the file, you don't need the `Cleanup` func, because the automatic `RemoveAll` succeeds.

Reply:
@bitfield If I don't add the file - then wouldn't the test pass regardless because FindFiles doesn't print directories. I was trying to show that there's a path inside the restricted dir that is not printed.

Reply:
Well, according to your test name, the _test_ is about whether or not `FindFiles` returns an error if it encounters one on its walk. If the error happens to be "can't list files in directory", then we hardly need to test that `FindFIles` doesn't return those files: how could it? Testing the impossible doesn't catch bugs (testing _assumptions_, on the other hand, is an excellent idea and we should always do it).

Reply:
@bitfield got it - thanks for the explanation! i removed the file for the test